File: tools/improve/tests/original/maudec/py/free-tuples.py
import enum

class Foo(enum.Enum):
	F = 0  # float, float
	G = 1  # float, float, bool
	H = 2  # float, bool, float, bool
	Err = 3  # 


def pi() -> float:
	return 3.141592653589793


def _u003du0060u005b_u0060u005d_i(X: float, Z: float, Y: float) -> bool:
	return abs((X - Y)) < Z


def first(a1: Foo) -> float:
	match a1:
		case (Foo.F, F1, _):
			return F1

		case (Foo.G, F1, _, _):
			return F1

		case (Foo.H, F1, _, _, _):
			return F1

File: tools/improve/tests/original/maudec/rust/free-tuples.rs


#[derive(Clone, PartialEq, PartialOrd, Debug)]
enum Foo {
	F(f64, f64),
	G(f64, f64, bool),
	H(f64, bool, f64, bool),
	Err,
}

fn pi() -> f64 {
	3.141592653589793
}

fn _u003du0060u005b_u0060u005d_(x: f64, z: f64, y: f64) -> bool {
	((x - y).abs() < z)
}

fn first(a1: &Foo) -> f64 {
	match a1 {
		Foo::F(f1, _) => *f1,
		Foo::G(f1, _, _) => *f1,
		Foo::H(f1, _, _, _) => *f1,
		_ => panic!("case not covered"),
	}
}

